hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockChecksumHelper.java

```diff
@@ -503,6 +503,7 @@ void compute() throws IOException {
           }
         } catch (IOException e) {
```

Comment:
It may be okay to just remove here instead of rethrowing exception because it is handled below too:
https://github.com/apache/hadoop/blob/f8769e0f4b917d9fda8ff7a9fddb4d755d246a1e/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java#L324

Reply:
Yes, we could just remove the try-catch block entirely and let the exception propagate upwards. However I wonder if it would make sense to catch it as we do now, but add some more information to the log for later debugging, eg the block causing the problem and the block index:

```
LOG.warn("Failed to get the checksum for block index {} and block{} in block group {}", idx, block, blockGroup, e);
throw(e);
```